ckan/logic/validators.py

```diff
@@ -858,3 +858,20 @@ def email_validator(value, context):
         if not email_pattern.match(value):
             raise Invalid(_('Email {email} is not a valid format').format(email=value))
     return value
+
+def one_of(value,context):
+   ''' Implementation of OneOf method
+       for Python Library Formencode
+   '''
+
+   if (value == '' or value == None):
```

Comment:
You might have `''` or `None` in your list of allowed values, so this should not raise an exception.